src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -638,20 +638,18 @@ modifyIndices = undefined
 -- /Pre-release/
 modify :: forall m a. (MonadIO m, Storable a)
     => Array a -> (a -> a) -> m ()
-modify Array{..} f = liftIO $ go arrStart
+modify Array{..} f = liftIO $
+    unsafeWithArrayContents arrContents arrStart $ \ptr -> do
+        let elemSize = sizeOf (undefined :: a)
+        go ptr elemSize
```

Comment:
We do not need to pass `elemSize` as it is statically determined, you can compute it inside `go` itself.